frida_tools/tracer.py

```diff
@@ -582,6 +601,9 @@ def _create_trace_script(self):
 var timer = null;
 
 rpc.exports = {
+    init: function (stage, parameters) {
+        state.userdata = %{userdata}s;
```

Comment:
```suggestion
        parameters = Object.freeze(params);
```

And adding `var parameters = {};` right before `var state = {};`. The thinking is that `parameters` should be treated as immutable input, and `state` is for mutable data only.